extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -446,13 +452,31 @@ export class AmpConsent extends AMP.BaseElement {
     if (!isExperimentOn(this.win, 'amp-consent-geo-override')) {
       return this.getConsentRequiredPromiseLegacy_();
     }
-    if (typeof this.consentConfig_['consentRequired'] === 'boolean') {
-      return Promise.resolve(this.consentConfig_['consentRequired']);
-    }
-    return this.getConsentRemote_().then(consentInfo => {
-      const remoteResponse = consentInfo['consentRequired'];
-      return typeof remoteResponse === 'boolean' ? remoteResponse : true;
-    });
+    return this.consentStateManager_
+      .getConsentInstanceInfo()
+      .then(storedInfo => {
+        if (hasStoredValue(storedInfo)) {
+          return Promise.resolve(false);
+        }
+        if (typeof this.consentConfig_['consentRequired'] === 'boolean') {
+          return Promise.resolve(this.consentConfig_['consentRequired']);
+        }
+        return this.getConsentRemote_().then(consentInfo => {
+          const consentRequired = consentInfo['consentRequired'];
+          const consentState = convertValueToState(
+            consentInfo['consentStateValue']
+          );
+          // Local storage empty, consent required, and not state is not unknow,
+          // then use the state given.
+          if (
```

Comment:
can this be handled in `syncConsentStringAndStateValue_`?

Reply:
good point